doc/kubernetes/modules/ROOT/pages/util/custom-image-for-keycloak.adoc

```diff
@@ -0,0 +1,64 @@
+= Using a custom Keycloak image for deployment in Kubernetes
+:navtitle: Using a custom Keycloak image
+:description: Use a custom image an external registry or a locally built image and run this in Kubernetes.
+
+{description}
+A running version of xref:installation.adoc[Keycloak within Minikube] is a prerequisite.
+
+== Overview
+
+Depending on the operator being used, it will pull a default image from the Keycloak's Quay.io registry.
+When running the nightly operator of Keycloak, it uses the nightly build of Keycloak.
+A specific version of the operator like, for example, 20.0.0, will use the version 20.0.0 of Keycloak.
+
+== Use a custom image available in a remote registry
+
+To change this, edit the environment file `.env` to contain the following:
+
+.Example entry in the `.env` file
+----
+KC_KEYCLOAK_IMAGE=quay.io/keycloak/keycloak:20.0.1
+----
+
+== Building a custom Keycloak image and using it
+
+. Check out https://github.com/keycloak/keycloak[Keycloak's Git repository].
+. Build using `mvn clean install -DskipTests` to create a `keycloak-999-SNAPSHOT.tar.gz` in folder `/quarkus/container`.
+. Configure the environment.
++
+.Example entry in the `.env` file
+----
+KC_KEYCLOAK_IMAGE=localhost/keycloak:local
+----
++
+To learn more about the `.env` file, see xref:customizing-deployment.adoc[].
+. Build the container, either with podman or with docker.
++
+.When running podman
+[source,bash]
+----
+cd quarkus/container
+cp ../dist/target/keycloak-*.tar.gz .
+podman build --build-arg KEYCLOAK_DIST=$(ls keycloak-*.tar.gz) . -t keycloak:local
+podman image save keycloak:local | minikube image load --overwrite=true -
+----
++
+.When running docker
```

Comment:
@ahus1 

I tried with the docker and I seem to end up in the below error with the Keycloak pod not able to pull the local image properly.

attached is the console output of me trying the steps for tagging the image locally using docker.
[docker_image_tag_full_console_log.txt](https://github.com/keycloak/keycloak-benchmark/files/10025342/docker_image_tag_full_console_log.txt)

Reply:
Task output confirming the usage of the .env variable `keycloakImage=localhost/keycloak:local keycloak` 

```
[keycloak] deployment.apps/keycloak-operator configured
task: [keycloak] helm upgrade --install keycloak --set hostname=192.168.39.188.nip.io --set db-pool-initial-size=50 --set db-pool-min-size=50 --set db-pool-max-size=50 --set storage=jpa --set database=postgres --set keycloakImage=localhost/keycloak:local keycloak

[keycloak] Release "keycloak" has been upgraded. Happy Helming!
[keycloak] NAME: keycloak
[keycloak] LAST DEPLOYED: Wed Nov 16 15:00:22 2022
[keycloak] NAMESPACE: default
[keycloak] STATUS: deployed
[keycloak] REVISION: 14
[keycloak] TEST SUITE: None
[keycloak] NOTES:
[keycloak] Keycloak will be available on https://keycloak.192.168.39.188.nip.io
[keycloak]   user: admin
[keycloak]   password: admin
[keycloak] sqlpad will be available on http://sqlpad.192.168.39.188.nip.io
[keycloak]   user: admin
[keycloak]   password: admin
[keycloak] Cryostat will be available on https://cryostat.192.168.39.188.nip.io
[keycloak] Connect to PostgreSQL on 192.168.39.188.nip.io:30009
[keycloak]   user: keycloak
[keycloak]   password: pass
[keycloak]   JDBC URL: jdbc:postgresql://192.168.39.188.nip.io:30009/keycloak
task: [keycloak] bash -c ' if [ "postgres" != "none" ]; then kubectl get deployment/postgres -n keycloak -o=jsonpath="{.spec}" > .task/status-keycloak-db.json; else echo "none" > .task/status-keycloak-db.json; fi'

task: [keycloak] bash -c 'kubectl get pods -A | grep -E "(BackOff|Error)" | tr -s " " | cut -d" " -f1-2 | xargs -r -L 1 kubectl delete pod -n'
task: [keycloak] bash -c 'sleep 1'
task: [keycloak] bash -c ./isup.sh
[keycloak] https://jaeger.192.168.39.188.nip.io/ is up

```

Reply:
the pod level event indicating the absence of the `localhost/keycloak:local` image
```
Events:
  Type     Reason             Age                   From     Message
  ----     ------             ----                  ----     -------
  Warning  ErrImageNeverPull  35s (x698 over 150m)  kubelet  Container image "localhost/keycloak:local" is not present with pull policy of Never

```

Reply:
image is listed in the docker images

```
$ docker images
REPOSITORY                                               TAG            IMAGE ID       CREATED         SIZE
keycloak                                                 local          6d53012ccade   3 hours ago     555MB

```

Reply:
Unfortunately I wasn't able to test this locally due to the lack of docker installed on my local machine. I'd like to try out some variants with you to make this work for both podman and docker in our call later today.